### ISSUE TITLE
ASP.NET Core template app once again does not run

### ISSUE THREAD

**Issue by jmoerdyk** [aurelia/cli#1014]
**I'm submitting a bug report**

* **Library Version:**
1.0.0-beta.9, 1.0.0-beta.8

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
8.14.0

* **Yarn Version:**
1.12.3

* **Browser:**
all 

* **Language:**
TypeScript

* **Loader/bundler:**
Webpack

**Current behavior:**
It looks like the webpack.config changes to the output bundles in 1.0.0-beta.8 have once again prevented the ASP.Net Core template app from running.  It appears that the Views/Home/Index.cshtml in the template needs to be updated with the correct output bundle file names.

Related issue: https://github.com/aurelia/cli/issues/979

* **What is the expected behavior?**
Use `au new` to create a Webpack/HTTP 1.1/ASP.NET Core/Typescript project
Start the project with `dotnet run`
If you open the app in the web browser, it just shows the text "Loading...", but the app does not start.  In the error console is the warning 

> Loading failed for the <script> with source “https://localhost:5001/dist/app.bundle.js”.

The source of the loaded page is:
```
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Home Page - Aurelia</title>
        <link rel="stylesheet" href="/dist/vendor.css">
    </head>
    <body>
        <div aurelia-app="main">Loading...</div>
       <script type="text/javascript" src="/dist/app.bundle.js"></script>
    </body>
</html>

```
Whereas the output file bundles are now:
```
01/09/2019  09:35 AM             9,152 app~d0ae3f07.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM            31,665 runtime~app.bundle.js
01/09/2019  09:35 AM           458,420 vendors~01be3b92.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           483,169 vendors~02227409.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           175,333 vendors~1f20a385.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           279,275 vendors~50e8d500.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           192,382 vendors~556c66f2.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM            44,495 vendors~5ea1390f.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           503,025 vendors~7e79ec10.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           258,675 vendors~814c0205.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM            69,847 vendors~ec8c427e.e331153e4bb2967ada4c.chunk.js
01/09/2019  09:35 AM           368,772 vendors~ecff2e3d.e331153e4bb2967ada4c.chunk.js
```

* **What is the motivation / use case for changing the behavior?**
To have the cli scaffolded app run out of the box

**Comment by beatsm** [comment]
Beta 9 also does not work vendor.css referenced in _layout is missing from the dist folder

**Comment by jlipford** [comment]
Running "au run" instead of "dotnet run" works and the references work.  Still looking at what to change to get dotnet to run properly as well

**Comment by jmoerdyk** [comment]
You'd need to update the` asp-net-src-include` attributes in the `@scripts` section of `lib/resources/content/javascriptservices/Views/Home/Index.cshtml` to match the new filename patterns to be something more like:
```
    <script type="text/javascript" asp-src-include="~/dist/runtime*.bundle.js" asp-append-version="true"></script>
    <script type="text/javascript" asp-src-include="~/dist/vendor*.chunk.js" asp-append-version="true"></script>
    <script type="text/javascript" asp-src-include="~/dist/app*.chunk.js" asp-append-version="true"></script>  
```

**Comment by dtaalbers** [comment]
What @jmoerdyk proposed as a solution worked for me. `Index.cshtml` now looks like this for me and it runs.

```
@{
    ViewData["Title"] = "Home Page";
}

<div aurelia-app="main">Loading...</div>

@section scripts {
<script type="text/javascript" asp-src-include="~/dist/runtime*.bundle.js" asp-append-version="true"></script>
<script type="text/javascript" asp-src-include="~/dist/vendor*.chunk.js" asp-append-version="true"></script>
<script type="text/javascript" asp-src-include="~/dist/app*.chunk.js" asp-append-version="true"></script>  
}
```

I had the same problem that @beatsm was having. But I just removed the reference in to `vendor.css` `_Layout.cshtml`. The sass configured in the aurelia project still works as expected. Also tried to publish the project and that worked fine to!

**Comment by jlipford** [comment]
While the script files are still loading the app is not initialized with "Hello World!" nor is the vendor.css building or any scss compiling.....

**Comment by jlipford** [comment]
> While the script files are still loading the app is not initialized with "Hello World!" nor is the vendor.css building or any scss compiling.....

My issue was that I kept the original <environment> tags and all 3 levels of scripts as denoted by @dtaalbers  above are needed for the app to render and scss to process.  All fixed once I put in the direct scripts.

**Comment by beatsm** [comment]
Interesting that the app now runs with the above suggestions but unfortunately hmr does not.

Ignored an update to unaccepted module app -> 5jyU -> GmYv -> 0
This is usually because the modules which have changed (and their parents) do not know how to hot reload themselves. See https://webpack.js.org/concepts/hot-module-replacement/ for more details.
[HMR]  - ./src/app.ts
[HMR] Reloading page

Did this ever work?

**Comment by dtaalbers** [comment]
@beatsm It is working for me out of the box. Is it correctly configured for you in the `Startup.cs`? E.g.:

```
 public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseWebpackDevMiddleware(new WebpackDevMiddlewareOptions
                {
                    HotModuleReplacement = true,
                    ConfigFile = "webpack.netcore.config.js",
                    HotModuleReplacementClientOptions = new Dictionary<string, string>
                    {
                        {"reload", "true"}
                    }
                });
            }
            else
            {
                app.UseExceptionHandler("/Home/Error");
                app.UseHsts();
            }
...
```

**Comment by airboss001** [comment]
Decided to play with the new CLI (v1.0.0-beta.12), and creating a brand new dotnet project states to run 'dotnet run'. _That is failing to find the app bundle._ 

'au run' works but that is not what the instructions state to do. 

Till it works, I would suggest changing the output instructions to one that actually works to prevent frustration as well as having to blow time trying to figure out why its not working.

**Comment by artmasa** [comment]
`au run` works because its actually creating an `index.html` in the `dist` folder. and not using the `Index.cshtml` to render the page and also is not using dotnet to serve it.
The suggested changes should be made to `Index.cshtml` to get it working again as suggested by @jmoerdyk 
Any progress on this? I got my project going but it's going to discourage the community to have these kind of problems.

**Comment by roddharris** [comment]
For those who are having issues with HMR, I have found the following to work as a temporary fix.  

1. Do not make the changes to the `Index.cshtml` file as people proposed.  For Development, be sure that you keep the line `<script type="text/javascript" src="~/dist/app.bundle.js" asp-append-version="true"></script>
2. Edit the `webpack.config.js` file and remove the entire `optimization` section.

Doing this seemed to solve both the javascript loading issue (although you may still need to remove the loading of `vendor.css` file for a clean load) and also resolved the HMR via dotnet issue.

One disclaimer, I am anything but a webpack expert so I don't honestly know all the implications of this little hack.  For me, everything seems to be working smoothly.

I'm hoping the CLI team is able to resolve the templates for .NetCore to get things working without having to make these additional changes.

**Comment by jmoerdyk** [comment]
I'm wondering if there are any changes that can be made to the `webpack.netcore.config.js` that would work with the optimized chunks, or if there is a way to disable the optimized chunking for development and leave it for production.  Then we can update the script references in `Index.cshtml` accordingly.  Unfortunately, I don't know enough about webpack or the .NET JavaScript Services WebpackDevMiddleware to know if that's possible, or how to do it.

**Comment by jmoerdyk** [comment]
This appears to be fixed as of 1.0.0 for me.
